File: Maximum_Depth_of_Binary_Tree.cpp
class Solution {
public:
int maxDepth(TreeNode* root) {//resolved(31/3/2024)//
     //min. runtime(9ms.)T(C(N))=O(N) and S(C(N)==O(N))) as it requires non contiguous memory allocation recursively
        if(!root) return 0;//root initalize
        int maxLeft = maxDepth(root->left);// left subtree declare
        int maxRight = maxDepth(root->right);// declare right subtree
        return max(maxLeft, maxRight)+1;}};//printing maximum left and right subtree Depth Value
 

File: Maximum_Depth_of_Binary_Tree.cs
public class Solution {
    public int MaxDepth(TreeNode root) {
        //Optimal Soln (T(C(N))=O(N) and S(C(N)==O(N))) as it requires non contiguous(dynamic) memory allocation recursively
        if(root==null)return 0;//initalizing root
        int leftree=MaxDepth(root.left);int rightree=MaxDepth(root.right);// left and right subtree declare
        return Math.Max(leftree,rightree)+1;}}//printing maximum left and right subtree Depth Val
